.github/workflows/master--lint-unit-build-and-publish-images.yml

```diff
@@ -0,0 +1,138 @@
+
+      - name: Build dependencies
+        run: yarn build:dependencies
+
+      - name: Lint
+        run: yarn ${{matrix.app}} lint
+
+      - name: Test
+        run: yarn ${{matrix.app}} test-cov
+
+      - name: Upload test results artifacts
+        uses: actions/upload-artifact@v2
+        if: ${{ success() || failure() }}
+        with:
+          name: test-results-${{matrix.app}}
+          path: ${{matrix.app}}/test-results
+          retention-days: 14
+
+      - name: Upload test coverage artifacts
+        uses: actions/upload-artifact@v2
+        if: ${{ success() || failure() }}
+        with:
+          name: test-coverage-${{matrix.app}}
+          path: ${{matrix.app}}/test-coverage
+          retention-days: 14
+
+  build-and-publish-images:
+    runs-on: ubuntu-latest
+    needs: [lint-and-test]
+    strategy:
+      matrix:
+        app: [designer, runner]
+    name: build-and-publish-${{matrix.app}}-image
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: "12.x"
+
+      - name: Docker compose pull
+        run: docker-compose pull
+
+      - name: Docker layer caching MONTH key
+        # MONTH key is used to rotate docker-layer-caching monthly to avoid the layers dangle issues
+        # see https://github.com/satackey/action-docker-layer-caching/issues/55
+        # if you change this, please remember to apply the same changes to branch--lint-unit-and-smoke-test.yml
+        # so master branch cache can be shared with branches
+        run: echo "MONTH=$(date +%m)" >> $GITHUB_ENV
+
+      - name: Docker layer caching
+        uses: satackey/action-docker-layer-caching@v0.0.11
+        continue-on-error: true
+        with:
+          key: digital-form-builder-${{ env.MONTH }}-{hash}
+          restore-keys: |
+            digital-form-builder-${{ env.MONTH }}
+
+      - name: Docker compose build ${{ matrix.app }}
+        run: |
+          LAST_TAG='${{ github.run_number }}-rc'
+          LAST_COMMIT='${{ github.sha }}'
+          docker-compose build
+
+      - name: Check images
+        run: |
+          docker image ls
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GHCR_PAT }}
+
+      - name: Tag image
+        run: |
+          docker tag digital-form-builder-${{matrix.app}} ghcr.io/xgovformbuilder/digital-form-builder-${{matrix.app}}:latest
+          docker tag digital-form-builder-${{matrix.app}} ghcr.io/xgovformbuilder/digital-form-builder-${{matrix.app}}:${{ github.run_number }}-rc
+
+      - name: Push Docker images
+        run: |
+          docker push ghcr.io/xgovformbuilder/digital-form-builder-${{matrix.app}}:latest
+          docker push ghcr.io/xgovformbuilder/digital-form-builder-${{matrix.app}}:${{ github.run_number }}-rc
```

Comment:
💩 because this is a "new action" it's going to start incrementing from 0 again, so we'll have tags like "111-rc" and "0-rc" for the same SHA. 

some things you can try
- use publish-images.yml
- change the name to match what was in publish-images (doubt this would work)

Reply:
Jsut to register, we agreed to change the tag to `1.0.${{github.run_number}}-rc`, solving the current issue, then once SemVer lands we can go from `1.1.X`